streams/src/main/java/org/apache/kafka/streams/state/StreamsMetadata.java

```diff
@@ -37,32 +38,40 @@
      * operations.
      */
     public final static StreamsMetadata NOT_AVAILABLE = new StreamsMetadata(new HostInfo("unavailable", -1),
-                                                                            Collections.emptySet(),
-                                                                            Collections.emptySet());
+                                                                                Collections.<String>emptySet(),
```

Comment:
This is a public class and hence any changes to it would be treated as a major change that requires a KIP. However, I'm wondering if exposing the `taskIds` are really necessary? See my comments above.